internal/migrations/001-common-actions.sql

```diff
@@ -364,6 +370,90 @@ CREATE OR REPLACE ACTION get_metadata(
        LIMIT $limit OFFSET $offset;
 };
 
+/**
+ * get_latest_metadata: Retrieves the latest metadata for a stream.
+ */
+CREATE OR REPLACE ACTION get_latest_metadata(
+    $data_provider TEXT,
+    $stream_id TEXT,
+    $key TEXT,
+    $ref TEXT
+) PUBLIC view returns table(
+    value_i INT,
+    value_f NUMERIC(36,18),
+    value_b BOOL,
+    value_s TEXT,
+    value_ref TEXT
+) {
+    for $row in get_metadata($data_provider, $stream_id, $key, $ref, 1, 0, 'created_at DESC') {
+        RETURN NEXT $row.value_i, $row.value_f, $row.value_b, $row.value_s, $row.value_ref;
+    }
+};
+
+/**
+ * get_latest_metadata_int: Retrieves the latest metadata value for a stream.
+ */
+CREATE OR REPLACE ACTION get_latest_metadata_int(
```

Comment:
these were implemented only to make easier to avoid errors on the schema development. I still recommend using only get metadata externally, as there it's easier to deduplicate code